tests/jax/components/building/environments_test.py

```diff
@@ -0,0 +1,219 @@
+import functools
+from types import SimpleNamespace
+from typing import Any
+
+import numpy as np
+import pytest
+
+from mava import MAEnvironmentSpec, specs
+from mava.components.jax.building.environments import (
+    EnvironmentSpec,
+    EnvironmentSpecConfig,
+    ExecutorEnvironmentLoop,
+    ExecutorEnvironmentLoopConfig,
+    ParallelExecutorEnvironmentLoop,
+)
+from mava.core_jax import SystemBuilder
+from mava.systems.jax import Builder
+from mava.utils.environments import debugging_utils
+from mava.utils.sort_utils import sort_str_num
+
+
+class AbstractExecutorEnvironmentLoop(ExecutorEnvironmentLoop):
+    """Implement abstract methods to allow testing of class"""
+
+    def on_building_executor_environment_loop(self, builder: SystemBuilder) -> None:
+        """Do nothing. Just implement abstract method"""
+        pass
+
+
+@pytest.fixture
+def test_environment_spec() -> EnvironmentSpec:
+    """Pytest fixture for environment spec"""
+    environment_factory = functools.partial(
+        debugging_utils.make_environment,
+        env_name="simple_spread",
+        action_space="discrete",
+        num_agents=10,
+    )
+
+    config = EnvironmentSpecConfig(environment_factory=environment_factory)
+    test_environment_spec = EnvironmentSpec(config=config)
+    return test_environment_spec
+
+
+@pytest.fixture
+def test_executor_environment_loop() -> ExecutorEnvironmentLoop:
+    """Pytest fixture for executor environment loop"""
+    config = ExecutorEnvironmentLoopConfig(should_update=False)
+    return AbstractExecutorEnvironmentLoop(config=config)
+
+
+@pytest.fixture
+def test_parallel_executor_environment_loop() -> ParallelExecutorEnvironmentLoop:
+    """Pytest fixture for executor environment loop"""
+
+    class ExecutorStatsWrapperClass:
```

Comment:
How about making this class inherits from `EnvironmentLoopStatisticsBase` so that you won't need `#type: ignore` in line 63

Reply:
Thanks @OmaymaMahjoub. That's a good suggestion, but I think it might be quite involved, since I have to call to the __init__ of `EnvironmentLoopStatisticsBase`, which requires an actual environment loop. Happy to go for this if you'd like me to though?

Reply:
Oh, I see yeah it makes sense that it needs to be a similar initiator if we do that, yeah as you like we can keep it like that, other than that I think it is a good PR thank you @mmorris44 !